evennia/prototypes/tests.py

```diff
@@ -312,11 +313,83 @@ def test_protfuncs(self):
+            mocked__obj_search.assert_not_called()
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("(#12345)", session=self.session), '(#12345)')
+            mocked__obj_search.assert_not_called()
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("obj(Char)", session=self.session), 'obj(Char)')
+            mocked__obj_search.assert_not_called()
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("objlist(#1)", session=self.session), 'objlist(#1)')
+            mocked__obj_search.assert_not_called()
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("dbref(Char)", session=self.session), 'dbref(Char)')
+            mocked__obj_search.assert_not_called()
+
+
+        # obj search happens
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("$objlist(#1)", session=self.session), ['#1'])
+            mocked__obj_search.assert_called_once()
+            assert ('#1',) == mocked__obj_search.call_args[0]
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("$obj(#1)", session=self.session), '#1')
+            mocked__obj_search.assert_called_once()
+            assert ('#1',) == mocked__obj_search.call_args[0]
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("$dbref(#1)", session=self.session), '#1')
+            mocked__obj_search.assert_called_once()
+            assert ('#1',) == mocked__obj_search.call_args[0]
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("$obj(Char)", session=self.session), '#6')
+            mocked__obj_search.assert_called_once()
+            assert ('Char',) == mocked__obj_search.call_args[0]
+
+
+        # bad invocation
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
+            self.assertEqual(protlib.protfunc_parser("$badfunc(#1)", session=self.session), '<UNKNOWN>')
+            mocked__obj_search.assert_not_called()
+
+        with mock.patch("evennia.prototypes.protfuncs._obj_search", wraps=protofuncs._obj_search) as mocked__obj_search:
```

Comment:
Looks good.

Reply:
Excellent @Griatch !

I just pushed 70eeb6f6 with the regex at module level as you suggested.

I think this PR can be merged and #1704 closed right after 🎉